public/modules/core/controllers/home.client.controller.js

```diff
@@ -164,27 +208,162 @@ angular.module('core').controller('HomeController', ['$scope', '$location', 'Aut
+                        {
+                            var temAlteration = [];
+                            _.each($scope.mutations, function(alterationItem){
+                                if(location[property].indexOf(alterationItem.mutationID) !== -1 )
+                                {
+                                    temAlteration.push(alterationItem);
+                                }
+                            });
+                            tempCriteria.push({type: property, value: temAlteration });
+                        }
+                        else
+                        {
+                            tempCriteria.push({type: property, value: location[property]});
+                        }
+
+                        if(property === 'tumor'){
+                            $scope.tumor = location[property];
+                        }
+                        else if(property === 'gene'){
+                            $scope.chosenGenes = location[property];
+                        }
+                        else if(property === 'mutation'){
+                            $scope.chosenMutations = location[property];
+                        }
+                        else if(property === 'country'){
+                            $scope.country = location[property];
                         }
-                    },
-                    function (error) {
-                        console.log('No hits in the mapping table');
                     }
-                );
+                    else
+                    {
+                        if(property === 'mutation')
+                        {
+                            var temAlteration = [];
+                            _.each($scope.mutations, function(alterationItem){
+                                if(location[property] === alterationItem.mutationID )
+                                {
+                                    temAlteration.push(alterationItem);
+                                }
+                            });
+                            tempCriteria.push({type: property, value: temAlteration });
+                        }
+                        else
+                        {
+                            tempCriteria.push({type: property, value: [location[property]]});
+                        }
+
+                        if(property === 'tumor'){
+                            $scope.tumor = [location[property]];
+                        }
+                        else if(property === 'gene'){
+                            $scope.chosenGenes = [location[property]];
+                        }
+                        else if(property === 'mutation'){
+                            $scope.chosenMutations = [location[property]];
+                        }
+                        else if(property === 'country'){
+                            $scope.country = [location[property]];
+                        }
+
```

Comment:
Could you try to reuse the code from line 248-274? No need to duplicate the code